images/win/scripts/SoftwareReport/SoftwareReport.Generator.ps1

```diff
@@ -15,279 +17,216 @@ Import-Module (Join-Path $PSScriptRoot "SoftwareReport.Java.psm1") -DisableNameC
-$toolsList = @(
-    (Get-7zipVersion),
-    (Get-Aria2Version),
-    (Get-AzCopyVersion),
-    (Get-BazelVersion),
-    (Get-BazeliskVersion),
-    (Get-BicepVersion),
-    (Get-CabalVersion),
-    (Get-CMakeVersion),
-    (Get-CodeQLBundleVersion),
-    (Get-DockerVersion),
-    (Get-DockerComposeVersion),
-    (Get-DockerComposeVersionV2),
-    (Get-DockerWincredVersion),
-    (Get-GHCVersion),
-    (Get-GitVersion),
-    (Get-GitLFSVersion),
-    (Get-InnoSetupVersion),
-    (Get-JQVersion),
-    (Get-KindVersion),
-    (Get-KubectlVersion),
-    (Get-MercurialVersion),
-    (Get-MinGWVersion),
-    (Get-NewmanVersion),
-    (Get-NSISVersion),
-    (Get-OpenSSLVersion),
-    (Get-PackerVersion),
-    (Get-PulumiVersion),
-    (Get-RVersion),
-    (Get-ServiceFabricSDKVersion),
-    (Get-StackVersion),
-    (Get-SVNVersion),
-    (Get-VSWhereVersion),
-    (Get-SwigVersion),
-    (Get-WinAppDriver),
-    (Get-WixVersion),
-    (Get-ZstdVersion),
-    (Get-YAMLLintVersion),
-    (Get-ImageMagickVersion)
-)
+# Software report
+$softwareReport = [SoftwareReport]::new($(Build-OSInfoSection))
+$optionalFeatures = $softwareReport.Root.AddHeader("Windows features")
+$optionalFeatures.AddToolVersion("Windows Subsystem for Linux (WSLv1):", "Enabled")
+$installedSoftware = $softwareReport.Root.AddHeader("Installed Software")
+
+# Language and Runtime
+$languageAndRuntime = $installedSoftware.AddHeader("Language and Runtime")
+$languageAndRuntime.AddToolVersion("Bash", $(Get-BashVersion))
+$languageAndRuntime.AddToolVersion("Go", $(Get-GoVersion))
+$languageAndRuntime.AddToolVersion("Julia", $(Get-JuliaVersion))
+$languageAndRuntime.AddToolVersion("Kotlin", $(Get-KotlinVersion))
+$languageAndRuntime.AddToolVersion("LLVM", $(Get-LLVMVersion))
+$languageAndRuntime.AddToolVersion("Node", $(Get-NodeVersion))
+$languageAndRuntime.AddToolVersion("Perl", $(Get-PerlVersion))
+$languageAndRuntime.AddToolVersion("PHP", $(Get-PHPVersion))
+$languageAndRuntime.AddToolVersion("Python", $(Get-PythonVersion))
+$languageAndRuntime.AddToolVersion("Ruby", $(Get-RubyVersion))
+
+# Package Management
+$packageManagement = $installedSoftware.AddHeader("Package Management")
+$packageManagement.AddToolVersion("Chocolatey", $(Get-ChocoVersion))
+$packageManagement.AddToolVersion("Composer", $(Get-ComposerVersion))
+$packageManagement.AddToolVersion("Helm", $(Get-HelmVersion))
+$packageManagement.AddToolVersion("Miniconda", $(Get-CondaVersion))
+$packageManagement.AddToolVersion("NPM", $(Get-NPMVersion))
+$packageManagement.AddToolVersion("NuGet", $(Get-NugetVersion))
+$packageManagement.AddToolVersion("pip", $(Get-PipVersion))
+$packageManagement.AddToolVersion("Pipx", $(Get-PipxVersion))
+$packageManagement.AddToolVersion("RubyGems", $(Get-RubyGemsVersion))
+$packageManagement.AddToolVersion("Vcpkg", $(Get-VcpkgVersion))
+$packageManagement.AddToolVersion("Yarn", $(Get-YarnVersion))
+
+$packageManagement.AddNode($(Build-PackageManagementEnvironmentTable))
```

Comment:
nitpick: could you please rework Build-PackageManagementEnvironmentTable to return table only so we can invoke it as:

```pwsh
$packageManagement.AddHeader("Environment variables").AddTable($(Build-PackageManagementEnvironmentTable))
```

Reply:
reworked